src/main/java/com/blackhearth/blockchain/wallet/BasicWalletFactory.java

```diff
@@ -0,0 +1,12 @@
+package com.blackhearth.blockchain.wallet;
+
+import com.blackhearth.blockchain.wallet.signature.SignatureUtils;
+
+public class BasicWalletFactory implements WalletFactory {
```

Comment:
Zmień nazwę klasy bo to sugeruje, że jest to wzorzec fabryki, a ewidentnie nie jest:
https://javadeveloper.pl/wzorzec-projektowy-fabryka/